element_array_ephys/ephys_no_curation.py

```diff
@@ -352,36 +354,101 @@ def make(self, key):
                     for key, shank, shank_col, shank_row in zip(
                         *electrode_query.fetch("KEY", "shank", "shank_col", "shank_row")
                     )
-                }
-
+                }  # electrode configuration
                 electrode_group_members = [
                     probe_electrodes[(shank, shank_col, shank_row)]
                     for shank, shank_col, shank_row, _ in spikeglx_meta.shankmap["data"]
-                ]
-            else:
-                raise NotImplementedError(
-                    "Processing for neuropixels probe model"
-                    " {} not yet implemented".format(spikeglx_meta.probe_model)
+                ]  # recording session-specific electrode configuration
+
+                # Compute hash for the electrode config (hash of dict of all ElectrodeConfig.Electrode)
+                electrode_config_hash = dict_to_uuid(
+                    {k["electrode"]: k for k in electrode_group_members}
+                )
+
+                electrode_list = sorted(
+                    [k["electrode"] for k in electrode_group_members]
                 )
+                electrode_gaps = (
+                    [-1]
+                    + np.where(np.diff(electrode_list) > 1)[0].tolist()
+                    + [len(electrode_list) - 1]
+                )
+                electrode_config_name = "; ".join(
+                    [
+                        f"{electrode_list[start + 1]}-{electrode_list[end]}"
+                        for start, end in zip(electrode_gaps[:-1], electrode_gaps[1:])
+                    ]
+                )
+
+                electrode_config_key = {"electrode_config_hash": electrode_config_hash}
+
+                # Insert into ElectrodeConfig
+                if not probe.ElectrodeConfig & electrode_config_key:
+                    probe.ElectrodeConfig.insert1(
+                        {
+                            **electrode_config_key,
+                            "probe_type": probe_type,
+                            "electrode_config_name": electrode_config_name,
+                        }
+                    )
+                    probe.ElectrodeConfig.Electrode.insert(
+                        {**electrode_config_key, **electrode}
+                        for electrode in electrode_group_members
+                    )
 
             self.insert1(
                 {
                     **key,
-                    **generate_electrode_config(probe_type, electrode_group_members),
+                    "electrode_config_hash": electrode_config_hash,
```

Comment:
Why removing this `generate_electrode_config` helper function only to have to duplicate this code block multiple time?